source/jormungandr/jormungandr/scenarios/helper_classes/streetnetwork_path.py

```diff
@@ -80,27 +93,105 @@ def __init__(
+    def poi_to_pt_object(self, poi):
+        return type_pb2.PtObject(poi=poi, uri=poi.uri, embedded_type=type_pb2.POI, name=poi.name)
+
+    def get_pt_objects(self, entry_point):
+        if include_poi_access_points(self._request, entry_point, self._mode):
+            for o in entry_point.poi.children:
+                yield self.poi_to_pt_object(o)
+        yield entry_point
+
+    def make_poi_access_points(self, fallback_type, best_dp_element):
+        entry_point = (
+            self._orig_obj if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK else self._dest_obj
+        )
+        if not include_poi_access_points(self._request, entry_point, self._mode):
+            return
+        via_poi_access = (
+            best_dp_element.origin
+            if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK
+            else best_dp_element.destination
+        )
+        _add_poi_access_point(fallback_type, via_poi_access, best_dp_element.response.journeys[0].sections)
+        _extend_with_via_poi_access(
+            best_dp_element.response,
+            fallback_type,
+            entry_point,
+            via_poi_access,
+            self._request.get('_asgard_language', "english_us"),
+        )
+        sections = best_dp_element.response.journeys[0].sections
+        sections[0].origin.CopyFrom(
+            self._orig_obj
+        ) if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK else sections[-1].destination.CopyFrom(
+            self._dest_obj
+        )
+
+    def finalize_direct_path(self, resp_direct_path):
+        if not resp_direct_path or not getattr(resp_direct_path.response, "journeys", None):
+            return response_pb2.Response()
+        self.make_poi_access_points(StreetNetworkPathType.BEGINNING_FALLBACK, resp_direct_path)
+        self.make_poi_access_points(StreetNetworkPathType.ENDING_FALLBACK, resp_direct_path)
+        return resp_direct_path.response
+
+    def build_direct_path(self):
+        best_direct_path = None
+        for origin in self.get_pt_objects(self._orig_obj):
+            for destination in self.get_pt_objects(self._dest_obj):
+                response = self._streetnetwork_service.direct_path_with_fp(
                     self._instance,
                     self._mode,
-                    self._orig_obj,
-                    self._dest_obj,
+                    origin,
+                    destination,
                     self._fallback_extremity,
                     self._request,
                     self._path_type,
                     self._request_id,
                 )
+                if not is_valid_direct_path(response):
+                    continue
+                if not best_direct_path:
+                    best_direct_path = Dp_element(origin, destination, response)
+                elif (
+                    response.journeys[0].durations.total < best_direct_path.response.journeys[0].durations.total
```

Comment:
we may obtain more than one journey with a direct_path in bike... let's talk about this...

Reply:
Done